tasks/src/tasks/sort.rs

```diff
@@ -0,0 +1,84 @@
+use std::num::NonZeroU32;
+
+use imap_types::{
+    command::CommandBody,
+    core::{Charset, Vec1},
+    extensions::sort::SortCriterion,
+    response::{Data, StatusBody, StatusKind},
+    search::SearchKey,
+};
+
+use super::TaskError;
+use crate::Task;
+
+#[derive(Clone, Debug)]
+pub struct SortTask {
+    sort_criteria: Vec1<SortCriterion>,
+    charset: Charset<'static>,
+    search_criteria: Vec1<SearchKey<'static>>,
+    uid: bool,
+    output: Vec<NonZeroU32>,
+}
+
+impl Task for SortTask {
+    type Output = Result<Vec<NonZeroU32>, TaskError>;
+
+    fn command_body(&self) -> CommandBody<'static> {
+        CommandBody::Sort {
+            sort_criteria: self.sort_criteria.clone(),
+            charset: self.charset.clone(),
+            search_criteria: self.search_criteria.clone(),
+            uid: self.uid,
+        }
+    }
+
+    fn process_data(&mut self, data: Data<'static>) -> Option<Data<'static>> {
+        if let Data::Sort(ids) = data {
```

Comment:
Grrrrrrr... Do we care about duplicate `Data::Sort`? I think we shouldn't but maybe emit a log again...? What do you think? (Maybe also a topic for later when we merged everything so far.)

Reply:
Emitting a log is good enough I would say. I pushed sth, let me know.